shopify/finance-dashboard-app-private-remix-ssr/api/models/shopifyOrder/actions/update.ts

```diff
@@ -0,0 +1,72 @@
+import { applyParams, save, ActionOptions, logger } from "gadget-server";
+import { preventCrossShopDataAccess } from "gadget-server/shopify";
+import { isFullPage } from "@notionhq/client";
+import { enqueueNotionJob, getNotionOrder } from "../utils";
+import { JSONValue } from "@gadget-client/finance-dashboard-app";
+import { api } from "gadget-server";
+
+export const run: ActionRun = async ({ params, record }) => {
+  applyParams(params, record);
+  await preventCrossShopDataAccess(params, record);
+  await save(record);
+};
+
+const updateOrderInNotion = async (
+  orderId: string,
+  totalPrice: string | null,
+  jsonValCOGS: JSONValue | null,
+  jsonValMargin: JSONValue | null
+) => {
+  const price = Number(totalPrice);
+  const costOfGoods = Number(jsonValCOGS);
+  const margin = Number(jsonValMargin);
+  const response = await getNotionOrder(orderId);
+  const updates: { id: string; pageId: string; value: number }[] = [];
+
+  for (const result of response.results) {
+    if (!isFullPage(result)) continue;
+
+    const typeProp = result.properties.Type;
+    const valueProp = result.properties.Value;
+    if (typeProp.type !== "select" || valueProp.type !== "number") continue;
+
+    let newValue: number | undefined;
+    if (typeProp.select?.name === "Sales") newValue = price;
+    if (typeProp.select?.name === "Cost of Goods") newValue = costOfGoods;
+    if (typeProp.select?.name === "Margin") newValue = margin;
+
+    if (newValue !== undefined && valueProp.number !== newValue)
+      updates.push({ id: orderId, pageId: result.id, value: newValue });
+  }
+
+  logger.info("*** updates ***");
+  logger.info(JSON.stringify(updates));
```

Comment:
Personal preference, but I like to use something like this:
`logger.info({ updates }, "*** update Notion order ***");
so that the `updates` object is stringified automatically and collapsed/expandable